distribution/docs/src/main/resources/_contents/_sources/developing-sources-contents.adoc

```diff
@@ -0,0 +1,113 @@
+=== Developing Sources
+
+Sources are components that enable ${branding} to talk to back-end services.
+They let ${branding} perform query and ingest operations on catalog stores and query operations on federated sources.
+
+==== Implement a Source Interface
+
+There are three types of sources that can be created.
+All of these types of sources can perform a query operation.
+Operating on queries is the foundation for all sources.
+All of these sources must also be able to return their availability and the list of content types currently stored in their back-end data stores.
+
+Catalog Provider:: `${ddf-branding}.catalog.source.CatalogProvider` _is used to communicate with back-end storage and allows for Query and Create/Update/Delete operations._
+Federated Source:: `${ddf-branding}.catalog.source.FederatedSource` _is used to communicate with remote systems and only allows query operations._
+Connected Source:: `${ddf-branding}.catalog.source.ConnectedSource` _is similar to a Federated Source with the following exceptions:_
+* _Queried on all local queries_
+* _`SiteName` is hidden (masked with the ${branding} sourceId) in query results_
+* _`SiteService` does not show this Source's information separate from ${branding}'s._
+Catalog Store:: `catalog.store.interface` is used to
+
+The procedure for implementing any of the source types follows a similar format:
+. Create a new class that implements the specified Source interface and `ConfiguredService`.
+. Implement the required methods.
+. Create an OSGi descriptor file to communicate with the OSGi registry. (Refer to <<_osgi_services,OSGi Services>>.)
+.. Import ${branding} packages.
+.. Register source class as service to the OSGi registry.
+. Deploy to ${branding}.
```

Comment:
It doesn't look like this is rendered correctly
![image](https://cloud.githubusercontent.com/assets/8041246/18290439/4ca17d32-7438-11e6-93cc-d644a1894bad.png)